### PR TITLE
[enhancement](disk) refine io error and report bad when disk is abnormal

### PR DESCRIPTION
**Pull request by dataroaring**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/tablet.h [hunk 1]
```diff
@@ -533,7 +533,13 @@ class Tablet : public BaseTablet {
     void gc_binlogs(int64_t version);
     Status ingest_binlog_metas(RowsetBinlogMetasPB* metas_pb);
 
-    inline void increase_io_error_times() { ++_io_error_times; }
+    inline void report_error(Status st) {
```
Comment (platoneko) [inline be/src/olap/tablet.h:536]: `const Status&`

**Review by platoneko — CHANGES_REQUESTED** [review]
Avoid Status copy.

**Comment by dataroaring** [comment]
run buildall

**Review by platoneko — APPROVED** [review]

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**